synapse/lib/storm.py

```diff
@@ -947,3 +948,65 @@ def getArgParser(self):
 
             if self.opts.join:
                 yield node, path
+
+class ScrapeCmd(Cmd):
+    '''
+    Use textual properties of existing nodes to find other easily recognizable nodes.
+
+    Examples:
+
+        inet:search:query | scrape
+        inet:search:query | scrape --refs
+    '''
+
+    name = 'scrape'
+
+    def getArgParser(self):
+        pars = Cmd.getArgParser(self)
+
+        pars.add_argument('--props', '-p', nargs='+', type=str, default=[],
+                          help='Specify relative properties to scrape')
+        pars.add_argument('--refs', '-r', default=False, action='store_true',
+                          help='Create refs to any scraped nodes from the source node')
+        pars.add_argument('-j', '--join', default=False, action='store_true',
+                          help='Include source nodes in the output of the command.')
+
+        return pars
+
+    async def execStormCmd(self, runt, genr):
+
+        async for node, path in genr:  # type: s_node.Node, s_node.Path
+
+            # repr all prop vals and try to scrape nodes from them
+            reprs = node.reprs()
+
+            props = [k for k in node.props.keys()]
+
+            # make sure any provided props are valid
+            for fprop in self.opts.props:
+                if fprop in props:
+                    continue
+                raise s_exc.BadOptValu('%r not a valid prop for %r' % (fprop, node.ndef))
+
+            # if a list of props haven't been specified, then default to ALL of them
+            if not self.opts.props:
+                self.opts.props = [k for k in node.props.keys()]
+
+            for prop in self.opts.props:
```

Comment:
It feels like we can dynamically generated this as the following:
```
props = set(node.props.keys())
if self.opts.props:
   props = props.intersection(self.opts.props)
for prop in props:
    ... do stuff
```

Reply:
this would allow a node whose form has a given prop; but the node itself is missing the prop, to pass through this command and not throw an error.